### PR TITLE
Add a prompt to allow users to enter the information about an existing database

### PR DESCRIPTION
**Pull request by wangxf123456**
I added 3 new prompts.

"NewDatabaseInformationPrompt" will be used to collect information for the new database. Previously we only ask the user for database password, and use "postgres" for database user name. In reality people rarely use "postgres" for authentication, so I also ask user for the database user name.

Users might want to use an existing database when running "cloudify" command. "ExistingDatabaseInformationPrompt" will be used to collect information for an existing database. It will collect the database username, password, public ip or host name, database name. I created an database instance with Amazon Relational Database Service, and these information should be enough to connect to an RDS instance.

When running "cloudify" command, I will ask users do they want to use an existing database or not, and use different prompt based on their answers.

NewDatabaseInformationPrompt will be like
```
[2/12] Do you want to create a new database or use an existing database for deployment? [y/N]: n
[2.a/12] Enter the master user name for the database: admin
[2.b/12] Enter a password for the database user "admin"
Password: 
Password (again): 
```

And ExistingDatabaseInformationPrompt will be like
```
[2/12] Do you want to create a new database or use an existing database for deployment? [y/N]: y
[2.a/12] Enter the public ip or host name of your database: mydbinstance3.ckukagg5823x.us-west-2.rds.amazonaws.com
[2.b/12] Enter the port number of your database or press Enter to use "5432":
[2.c/12] Enter the master user name for the database: admin
[2.d/12] Enter password for the database user "admin"
Password: 
Password (again): 
[2.e/12] Enter the name of your database: mydb
```

### CODE REVIEW

**Comment by wangxf123456** [comment]
@enriquejosepadilla I need your opinions for the implementation details so that I fix it early. I will add tests to this PR after it looks good to you.

**Comment by enriquejosepadilla** [comment]
1. Yes/no isnt clear for the user if they are using existing/new
2. It's worth discussing the fact you are using a dictionary that collects multiple pieces of information. It deviates from the "norm" , but the norm would clutter for a lot of small prompts.

File: django_cloud_deploy/cli/prompt.py [hunk 1]
```diff
@@ -838,6 +860,205 @@ def _validate(self, s):
             raise ValueError('The provided billing account does not exist.')
 
 
+class GroupingPrompt(TemplatePrompt):
```
Comment (enriquejosepadilla) [inline django_cloud_deploy/cli/prompt.py]: What are the benefits of making this a class that gets inherited vs a utility function?

File: django_cloud_deploy/cli/prompt.py [hunk 2]
```diff
@@ -838,6 +860,205 @@ def _validate(self, s):
...
+        database = database_arguments.get('database_name')
+        conn = None
+        try:
+            conn = psycopg2.connect(host=host,
+                                    database=database,
+                                    user=user,
+                                    password=password,
+                                    port=port)
+        except psycopg2.Error as e:
+            raise ValueError(
```
Comment (enriquejosepadilla) [inline django_cloud_deploy/cli/prompt.py]: Is it worth logging the values the user provided to help with debugging? (WIthout the password)

File: django_cloud_deploy/cli/prompt.py (hunk 1 above)
Comment (wangxf123456) [reply django_cloud_deploy/cli/prompt.py]: Right now this utility function is only used in the two database prompts. Maybe it is not necessary to make it visible to other classes. But I am fine to make it a utility function.

File: django_cloud_deploy/cli/prompt.py (hunk 2 above)
Comment (wangxf123456) [reply django_cloud_deploy/cli/prompt.py]: The values will be shown to the user when they fill in the values:

```
[2/12] Do you want to create a new database or use an existing database for deployment? [y/N]: y
[2.a/12] Enter the public ip or host name of your database: mydbinstance3.ckukagg5823x.us-west-2.rds.amazonaws.com
[2.b/12] Enter the port number of your database or press Enter to use "5432":
[2.c/12] Enter the master user name for the database: admin
[2.d/12] Enter password for the database user "admin"
Password: 
Password (again): 
[2.e/12] Enter the name of your database: mydb
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/google/home/wangxf/Desktop/code/django-cloud-deploy/venv/lib/python3.6/site-packages/psycopg2/__init__.py", line 130, in connect
    conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
psycopg2.OperationalError: FATAL:  password authentication failed for user "admin"
FATAL:  password authentication failed for user "admin"
```

**Pull request merged**